### PR TITLE
Remove SaveActor Fixup and Setup - depends on liblcf #399

### PR DESCRIPTION
**Pull request by fmatthew5876**
Depends on: https://github.com/EasyRPG/liblcf/pull/399

DynRPG Addresses
---------------------

| Address | Class | Function | Comment |
| -- | -- | -- | -- |
| 004B5330 | Actor | Reset | Resets all fields to default values |
| 004B6DD8 | Actor | SetTitle |
| 004B7004 | Actor | SetFace |
| 004B6D34 | Actor | SetName |
| 004B6E58 | Actor | GetCharset |
| 004B6E94 | Actor | GetCharsetId |
| 004B6EBC | Actor | GetCharsetTransparency |
| 004B6D88 | Actor | GetTitle |
| 004B6FDC | Actor | GetFaceId |
| 004B6FA0 | Actor | GetFaceName |
| 004B6CE4 | Actor | GetName |
| 004B5A48 | Actor | InitSaveChunk | Maps save chunk ids to fields

### CODE REVIEW

**Comment by fmatthew5876** [comment]
Rebased and retested this, including the minor battle test cleanup.

This is ready

**Review by fdelapena — APPROVED** [review]

File: src/game_actor.cpp [hunk 1]
```diff
@@ -58,62 +58,66 @@ Game_Actor::Game_Actor(int actor_id) {
...
-	RemoveInvalidData();
+	MakeExpList();
+	SetBattlePosition(GetOriginalPosition());
 
-	if (GetLevel() > 0) {
-		SetHp(GetMaxHp());
-		SetSp(GetMaxSp());
-		SetExp(exp_list[GetLevel() - 1]);
-	}
+	ChangeLevel(dbActor->initial_level, nullptr);
```
Comment (Ghabry) [inline src/game_actor.cpp:78]: Needs a test for COLORS: Lost Memories (there is a comment in this file somewhere) which uses Level 0.

**Comment by Ghabry** [comment]
btw the only reason why the title scene invokes the battle test is because newer engines do it this way.

**Comment by fmatthew5876** [comment]
> btw the only reason why the title scene invokes the battle test is because newer engines do it this way.

Given that many things could potentially depend on the title scene existing in the stack, this is a safer way to do it anyway.

File: src/game_actor.cpp (hunk 1 above)
Comment (fmatthew5876) [reply src/game_actor.cpp]: Retested and fixed a bug with this. Thanks for raising it.

**Review by Ghabry — APPROVED** [review]

**Comment by Ghabry** [comment]
jenkins: test this please

**Pull request merged**